src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -277,23 +297,21 @@ private String resolveJsModuleDirectory(String moduleAddress) {
 
     @Nullable
     private String resolveJsModuleFromRegistry(String moduleAddress) {
-      for (Map.Entry<String, ImmutableSet<String>> nodeModulesFolder :
-          nodeModulesRegistry.entrySet()) {
-        if (!this.path.startsWith(nodeModulesFolder.getKey())) {
+      for (String nodeModulesFolder : nodeModulesFolders) {
+        if (!toAbsoluteIdentifier(this.path).startsWith(nodeModulesFolder)) {
           continue;
         }
 
         // Load as a file
-        String fullModulePath = nodeModulesFolder.getKey() + "node_modules/" + moduleAddress;
+        String fullModulePath = nodeModulesFolder + "node_modules/" + moduleAddress;
         String loadAddress = resolveJsModuleFile(fullModulePath);
-        if (loadAddress != null) {
-          return loadAddress;
+        if (loadAddress == null) {
+          // Load as a directory
+          loadAddress = resolveJsModuleDirectory(fullModulePath);
         }
 
-        // Load as a directory
-        loadAddress = resolveJsModuleDirectory(fullModulePath);
         if (loadAddress != null) {
-          return loadAddress;
+          return removeAbsoluteIdentifierIndicator(loadAddress);
```

Comment:
Could you add a comment to say why you remove the absolute identifier indicator here?
Maybe in javadoc on the method.